lib/Basic/LangOptions.cpp

```diff
@@ -641,6 +651,17 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
     addPlatformConditionValue(PlatformConditionKind::PointerBitWidth, "_64");
   }
 
+  // Set the "_objectFileFormat" platform condition.
+  if (Target.isOSBinFormatMachO()) {
+    addPlatformConditionValue(PlatformConditionKind::ObjectFileFormat, "MachO");
+  } else if (Target.isOSBinFormatELF()) {
+    addPlatformConditionValue(PlatformConditionKind::ObjectFileFormat, "ELF");
+  } else if (Target.isOSBinFormatCOFF()) {
+    addPlatformConditionValue(PlatformConditionKind::ObjectFileFormat, "COFF");
+  } else if (Target.isOSBinFormatWasm()) {
+    addPlatformConditionValue(PlatformConditionKind::ObjectFileFormat, "Wasm");
+  }
```

Comment:
Can we rewrite this as:

```
addPlatformConditionValue(PlatformConditionKind::ObjectFileFormat,
  llvm::StringSwitch<llvm::Triple::ObjectFormatType>(Target.getObjectFormat())
    .Case(llvm::Triple::COFF, "COFF")
    .Case(llvm::Triple::ELF, "ELF")
    .Case(llvm::Triple::MachO, "MachO")
    .Case(llvm::Triple::Wasm, "Wasm"));
```

Reply:
Hm, not really, because we're not switching over strings here, but over the enum cases... I could do a regular switch, but that reads worse than the chain of ifs:
```
  StringRef FileFormatName;
  switch (Target.getObjectFormat()) {
  case llvm::Triple::COFF:
    FileFormatName = "COFF";
    break;
  case llvm::Triple::ELF:
    FileFormatName = "ELF";
    break;
  case llvm::Triple::MachO:
    FileFormatName = "MachO";
    break;
  case llvm::Triple::Wasm:
    FileFormatName = "Wasm";
    break;
  default:
    break;
  }
  if (!FileFormatName.empty()) {
    addPlatformConditionValue(PlatformConditionKind::ObjectFileFormat,
                              FileFormatName);
  }
```

Reply:
The covered switch would make it easier to identify new cases though I think. It is more that I am after a switch rather than a series of cascading `if` conditions.